html/includes/authentication/active_directory.inc.php

```diff
@@ -81,11 +83,20 @@ function auth_usermanagement() {
 }
 
 
-function adduser() {
-    // not supported so return 0
-    return 0;
+function adduser($username) {
+    // Check to see if user is already added in the database
+    if (!user_exists_in_db($username)) {
+        return dbInsert(array('username' => $username, 'user_id' => get_userid($username), 'level' => "0", 'can_modify_passwd' => 0, 'twofactor' => 0), 'users');
+    }
+    else {
+        return false;
+    }
 }
 
+function user_exists_in_db($username) {
+    $return = @dbFetchCell('SELECT COUNT(*) FROM users WHERE username = ?', array($username), true);
```

Comment:
Any reason why you're suppressing the dbFetchCell call with @? Just return the count and and check if it's > 0 maybe?